jenkins/printJarVersion.sh

```diff
@@ -23,10 +24,15 @@ function print_ver(){
     SERVER_ID=$5
     
     if [[ "$VERSION" == *"-SNAPSHOT" ]]; then
-        PREFIX=${VERSION%-SNAPSHOT}
-        TIMESTAMP=`grep -oP '(?<=timestamp>)[^<]+' < $REPO/maven-metadata-$SERVER_ID.xml`
-        BUILD_NUM=`grep -oP '(?<=buildNumber>)[^<]+' < $REPO/maven-metadata-$SERVER_ID.xml`
-        echo $TAG=$PREFIX-$TIMESTAMP-$BUILD_NUM$SUFFIX
+        # Collect snapshot dependency info only in Jenkins build, 
+        # In dev build, print 'SNAPSHOT' tag without time stamp, e.g.: cudf-0.17-SNAPSHOT.jar
+        if [ -n "$JENKINS_URL" ]; then
```

Comment:
Isn't this backwards?  Don't we want a timestamp when `JENKINS_URL` is set?  I suggest simplifying the logic to something like:
```
    if [[ "$VERSION" == *"-SNAPSHOT" && -n "$JENKINS_URL" ]]; then
        # Collect snapshot dependency info only in Jenkins build, 
        PREFIX=${VERSION%-SNAPSHOT}
        # List the latest SNAPSHOT jar file in the maven repo
        echo $TAG=`ls -t $REPO/$PREFIX-[0-9]*$SUFFIX | head -1 | xargs basename`
    else
        echo $TAG=$VERSION$SUFFIX
    fi
```
Then we'll just print the jar filename i a dev build but do the timestamp shenanigans when we have Jenkins.

Reply:
Agree!

Reply:
build